src/libtsduck/dtv/analysis/tsTr101Analyzer.cpp

```diff
@@ -0,0 +1,503 @@
+
+    stm.add(ts::UString::FromUTF8(name), count);
+    ts::json::Value& pids(stm.query(u"pids", true));
+    json(ts::UString::FromUTF8(name).data(), fn, stm, pids, _services);
+}
+
+bool is_pes(const ts::TR101_290Analyzer::ServiceContext& ctx) {return ctx._type == ts::TR101_290Analyzer::ServiceContext::Assigned;}
+bool is_pat(const ts::TR101_290Analyzer::ServiceContext& ctx) {return ctx._type == ts::TR101_290Analyzer::ServiceContext::Pat;}
+bool is_pmt(const ts::TR101_290Analyzer::ServiceContext& ctx) {return ctx._type == ts::TR101_290Analyzer::ServiceContext::Pmt;}
+bool is_table(const ts::TR101_290Analyzer::ServiceContext& ctx) {return ctx._type == ts::TR101_290Analyzer::ServiceContext::Pmt || ctx._type == ts::TR101_290Analyzer::ServiceContext::Pat;}
+
+ErrorState get_sync_byte_err(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.sync_byte_error, true};}
+ErrorState get_pat_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pat_err.count, is_pat(ctx), ctx.pat_err};}
+ErrorState get_pat_error2(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pat_err2.count,is_pat(ctx), ctx.pat_err2};}
+ErrorState get_cc_err(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.cc_error, true};}
+ErrorState get_pmt_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pmt_err.count, is_pmt(ctx), ctx.pmt_err};}
+ErrorState get_pmt_error2(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pmt_err2.count, is_pmt(ctx), ctx.pmt_err2};}
+ErrorState get_pid_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pid_err.count, !is_pmt(ctx) && !is_pat(ctx), ctx.pid_err};}
+ErrorState get_transport_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.transport_error, true};}
+ErrorState get_crc_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.crc_error, is_table(ctx)};}
+ErrorState get_pcr_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pcr_error, ctx.has_pcr};}
+ErrorState get_repetition_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pcr_repetition_err.count, ctx.has_pcr, ctx.pcr_repetition_err};}
+ErrorState get_discontinuity_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pcr_discontinuity_err.count, ctx.has_pcr, ctx.pcr_discontinuity_err};}
+ErrorState get_accuracy_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pcr_accuracy_err.count, ctx.has_pcr, ctx.pcr_accuracy_err};}
+ErrorState get_pts_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.pts_err.count, is_pes(ctx), ctx.pts_err};}
+ErrorState get_cat_error(const ts::TR101_290Analyzer::ServiceContext& ctx) {return {ctx.cat_error, true};}
+
+void ts::TR101_290Analyzer::report(std::ostream& stm, int& opt, Report& rep)
+{
+    stm << "Priority 1 Errors:\n";
+    // todo: TS_sync_loss may not be a valid test in IP-based systems and covered by Sync_byte_error.
```

Comment:
TS_sync_loss is defined as: 
> Loss of synchronization with consideration of hysteresis parameters

I am not entirely sure what this means, or if it even applies in file or IP-based formats. Any ideas?

Reply:
> I am not entirely sure what this means

Same for me...

Reply:
Okay I found some more information about this particular error:

> TS_sync_loss
The most important function for the evaluation of data from the MPEG-2 TS is the sync acquisition. The actual
synchronization of the TS depends on the number of correct sync bytes necessary for the device to synchronize and on the number of distorted sync bytes which the device cannot cope with. 

> It is proposed that five consecutive correct sync bytes (ISO/IEC 13818-1 [i.1], clause G.1) should be sufficient for sync acquisition, and two or more consecutive corrupted sync bytes should indicate sync loss. 

> After synchronization has been achieved the evaluation of the other parameters can be carried out.

Reply:
OK, so it seems the same topic as the 0x47 sync bytes. As I previously commented, this is a demodulation issue, not something that can be reliably tested in TSDuck. In the case of the `dvb` input plugin, the resync is done by the demod and we start receiving packets once the demod has resync. So, we can't really measure it. This must be measured by specialised equipment, or maybe by some SDR software using a RF receiver.

Reply:
Following up on this up in the earlier comment.